lib/niconico/api.rb

```diff
@@ -0,0 +1,46 @@
+# -*- coding: utf-8 -*-
+require 'json'
+
+class Niconico
+  class API
+    class ApiParseError < Exception; end
+
+    def initialize(parent, token=nil)
+      @parent = parent
+      @agent = parent.agent
+      @token = token || get_token
+    end
+
+    def get_token
+      page = @agent.get(Niconico::URL[:mymylist])
+      if page.search("script").map(&:inner_text).find{|x| /\tNicoAPI\.token/ =~ x }.match(/\tNicoAPI\.token = "(.+)";\n/)
+        @token = $1
+      else
+        raise ApiParseError, 'token can not be acquired'
```

Comment:
At here we're not using API, consider creating another exception class (`AcquiringTokenError` ?)